edx_sailthru/sailthru_content/services/catalog_api_service.py

```diff
@@ -81,5 +81,6 @@ def get_program_dictionary(self):
                 marketable_course_runs_only=1
             )
             for program in program_array:
-                self._programs_dictionary[program['uuid']] = program
+                if program['status'] == 'active':
```

Comment:
Instead of manually filter, maybe you should just call with a flag documented at https://github.com/edx/course-discovery/blob/master/course_discovery/apps/api/v1/views/programs.py#L49